tool/src/test/java/org/wildfly/security/tool/FileSystemEncryptRealmCommandTest.java

```diff
@@ -0,0 +1,137 @@
+    }
+
+    private void runCommand(String bulkConvertFile, int expectedStatus) {
+        String[] requiredArgs;
+        requiredArgs = new String[]{"--bulk-convert", bulkConvertFile};
+        executeCommandAndCheckStatus(requiredArgs, expectedStatus);
+    }
+
+    private void runCommandSilent(String inputLocation, String outputLocation, String fileSystemRealmName, int expectedStatus) {
+        String[] requiredArgsSilent;
+        requiredArgsSilent = new String[]{"--input-location", inputLocation, "--output-location", outputLocation, "--realm-name", fileSystemRealmName, "--silent"};
+        executeCommandAndCheckStatus(requiredArgsSilent, expectedStatus);
+    }
+
+    private void runCommandSilent(String bulkConvertFile, int expectedStatus) {
+        String[] requiredArgsSilent;
+        requiredArgsSilent = new String[]{"--bulk-convert", bulkConvertFile, "--silent"};
+        executeCommandAndCheckStatus(requiredArgsSilent, expectedStatus);
+    }
+
+    @Override
+    protected String getCommandType() {
+        return FileSystemEncryptRealmCommand.FILE_SYSTEM_ENCRYPT_COMMAND;
+    }
+
+    @Test
+    public void testHelp() {
+        String[] args = new String[]{"--help"};
+        executeCommandAndCheckStatus(args);
+    }
+
+    @Test
+    public void testBulk() throws Exception {
+        String descriptorFileLocation = "./target/test-classes/bulk-encryption-conversion-desc";
+        runCommand(descriptorFileLocation, 0);
+        String[] files = new String[]{"multiple-credential-types/O/OBWGC2LOKVZWK4Q.xml", "multiple-credential-types/O/ONQWY5DFMRKXGZLS.xml", "multiple-credential-types/O/OVZWK4RUGI.xml", "multiple-credential-types/M/MFXG65DIMVZFK43FOI.xml", "multiple-credential-types/M/MFZWQ4DBNY.xml", "multiple-credential-types/N/NZSXOU3BNR2GKZCVONSXEMQ.xml", "hash-encoding/O/B/OBSXE43PNYZA.xml", "hash-encoding/O/5/O5UWYZDGNR4TO.xml", "hash-encoding/O/V/OVZWK4RR.xml", "hash-encoding/M/J/MJXXSNA.xml", "hash-encoding/M/5/M5UXE3BV.xml", "hash-encoding/M/V/MVQXAOA.xml", "hash-encoding/N/J/NJRG643TGY.xml", "hash-encoding/N/F/NFSGK3TUNF2HSMY.xml", "hash-charset/O/B/OBSXE43PNYZA.xml", "hash-charset/O/5/O5UWYZDGNR4TO.xml", "hash-charset/O/V/OVZWK4RR.xml", "hash-charset/M/J/MJXXSNA.xml", "hash-charset/M/5/M5UXE3BV.xml", "hash-charset/M/V/MVQXAOA.xml", "hash-charset/N/J/NJRG643TGY.xml", "hash-charset/N/F/NFSGK3TUNF2HSMY.xml", "level-4/O/B/S/X/OBSXE43PNYZA.xml", "level-4/O/5/U/W/O5UWYZDGNR4TO.xml", "level-4/O/V/Z/W/OVZWK4RR.xml", "level-4/M/J/X/X/MJXXSNA.xml", "level-4/M/5/U/X/M5UXE3BV.xml", "level-4/M/V/Q/X/MVQXAOA.xml", "level-4/N/J/R/G/NJRG643TGY.xml", "level-4/N/F/S/G/NFSGK3TUNF2HSMY.xml"};
+        for (String file: files) {
+            if(!fileExists("target/test-classes/filesystem-encrypt/fs-encrypted-realms/"+file)){
+                throw new FileNotFoundException("Missing file from Bulk Descriptor File: " + file);
+            }
+        }
+    }
+
+    @Test
+    public void testSingleUser() throws Exception {
```

Comment:
In these tests, it would be good to try to instantiate a `FileSystemRealm` using the encrypted files and attempt to read an identity's credentials / attributes to make sure that works.

Reply:
Added test `testSingleUserAndVerify()`